usr/src/uts/common/fs/zfs/dbuf.c

```diff
@@ -3284,13 +3284,14 @@ dbuf_write_children_ready(zio_t *zio, arc_buf_t *buf, void *vdb)
 	dmu_buf_impl_t *db = vdb;
 	dnode_t *dn;
 	blkptr_t *bp;
-	uint64_t i;
+	int i;
```

Comment:
If epbs == 31, then you overflow this int.  I'm surprised lint doesn't whine about this.  Please either make it at least unsigned or explicitly uint32_t.